pkg/apis/configuration/validation/policy.go

```diff
@@ -159,13 +161,46 @@ func validateJWT(jwt *v1.JWTAuth, fieldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, validateRealm(jwt.Realm, fieldPath.Child("realm"))...)
 	}
 
-	if jwt.Secret == "" {
-		return append(allErrs, field.Required(fieldPath.Child("secret"), ""))
+	if jwt.Secret == "" && jwt.JwksURI == "" {
+		return append(allErrs, field.Required(fieldPath.Child("secret"), "either Secret or JwksURI must be present"))
 	}
+
+	if jwt.Secret != "" && jwt.JwksURI != "" {
+		return append(allErrs, field.Forbidden(fieldPath.Child("secret"), "only either of Secret or JwksURI can be used"))
+	}
+
+	if jwt.KeyCache != "" && jwt.JwksURI == "" {
+		return append(allErrs, field.Required(fieldPath.Child("jwksURI"), "jwksURI must be present when keyCache is used."))
+	}
+
 	allErrs = append(allErrs, validateSecretName(jwt.Secret, fieldPath.Child("secret"))...)
 
 	allErrs = append(allErrs, validateJWTToken(jwt.Token, fieldPath.Child("token"))...)
 
+	if jwt.JwksURI != "" {
+		allErrs = append(allErrs, validateURL(jwt.JwksURI, fieldPath.Child("jwksURI"))...)
+	}
+
+	if jwt.KeyCache != "" {
+		allErrs = append(allErrs, validateKeyCache(jwt.KeyCache, fieldPath.Child("keyCache"))...)
+	}
+
+	return allErrs
+}
+
+func validateKeyCache(name string, fieldPath *field.Path) field.ErrorList {
```

Comment:
can't you use `validateTime`?

Reply:
Thanks for pointing that out. I tried it, and all tests passed except one that testing for missing suffix. But it should actually be allowed because the [documentation](https://nginx.org/en/docs/syntax.html) states that "A value without a suffix means seconds." 

I have pushed a new commit. Hope it's okay for you @shaun-nx